.github/workflows/build-container.yml

```diff
@@ -4,6 +4,10 @@ on:
     branches:
       - 'main'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
imho it would be better to use
```
    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```
and not cancel the main branch workflows, to prevent the red X